account/client.go

```diff
@@ -18,107 +16,16 @@ func New(params *stripe.AccountParams) (*stripe.Account, error) {
 	return getC().New(params)
 }
 
-func writeAccountParams(
-	params *stripe.AccountParams, body *stripe.RequestValues,
-) {
-	if len(params.Country) > 0 {
-		body.Add("country", params.Country)
-	}
-
-	if len(params.Email) > 0 {
-		body.Add("email", params.Email)
-	}
-
-	if params.DebitNegativeBal {
-		body.Add("debit_negative_balances", strconv.FormatBool(true))
-	} else if params.NoDebitNegativeBal {
-		body.Add("debit_negative_balances", strconv.FormatBool(false))
-	}
-
-	if len(params.DefaultCurrency) > 0 {
-		body.Add("default_currency", params.DefaultCurrency)
-	}
-
-	if params.ExternalAccount != nil {
-		if len(params.ExternalAccount.Token) > 0 {
```

Comment:
How is your code handling those cases now? Before if I passed token *and* other things it still worked and used the tokens. Now that code will completely fail because it will pass everything at once.

I don't think it's bad per-se but it's a big change that's hard to notice and catch at build time so I wanted to flag.

Reply:
Yeah, so the current implementation will recurse into `ExternalAccount` and possibly serialize each of its members in the same way that it did for its parent.

Kyle flagged something similar before too. Even though nothing has changed interface-wise, we're still going to have do a major version bump for this because we're just going to be encoding more fields than we did before. Certain mistakes that might've been ignored will now be bubbled up to you as an API error (hopefully the number of these cases will be minimal, but it's still something we'll have to look out for).

Reply:
Agreed but we have made so many major bump that it does not mean "much" for stripe-go compared to other libraries.